clang/test/dpct/debug_test/all_public_dump/generated_schema.hpp.cuda.ref

```diff
@@ -1,190 +1,311 @@
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  template <> class DataSer<Point3DExt> {
-//CHECK:  public:
-//CHECK:    static void dump(dpct::experimental::detail::json_stringstream &ss, Point3DExt &value,
-//CHECK:                     dpct::experimental::StreamType stream) {
-//CHECK:      Point3DExt_codepin& temp = reinterpret_cast<Point3DExt_codepin&>(value);
-//CHECK:      dpct::experimental::detail::DataSer<Point3DExt_codepin>::dump(ss, temp, stream);
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  template <> class DataSer<Point3D_codepin<float>> {
-//CHECK:  public:
-//CHECK:    static void dump(dpct::experimental::detail::json_stringstream &ss, Point3D_codepin<float> &value,
-//CHECK:                     dpct::experimental::StreamType stream) {
-//CHECK:      ss << "{\"Type\":\"Point3D<float>\",\"Data\":[";
-//CHECK:      ss << "{\"x\":";
-//CHECK:      dpct::experimental::detail::DataSer<float>::dump(ss, value.x, stream);
-//CHECK:      ss << "},";
-//CHECK:      ss << "{\"y\":";
-//CHECK:      dpct::experimental::detail::DataSer<float>::dump(ss, value.y, stream);
-//CHECK:      ss << "},";
-//CHECK:      ss << "{\"z\":";
-//CHECK:      dpct::experimental::detail::DataSer<float>::dump(ss, value.z, stream);
-//CHECK:      ss << "}";
-//CHECK:      ss << "]}";
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  template <> class DataSer<Point3D<float>> {
-//CHECK:  public:
-//CHECK:    static void dump(dpct::experimental::detail::json_stringstream &ss, Point3D<float> &value,
-//CHECK:                     dpct::experimental::StreamType stream) {
-//CHECK:      Point3D_codepin<float>& temp = reinterpret_cast<Point3D_codepin<float>&>(value);
-//CHECK:      dpct::experimental::detail::DataSer<Point3D_codepin<float>>::dump(ss, temp, stream);
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  template <> class DataSer<Point2D_codepin> {
-//CHECK:  public:
-//CHECK:    static void dump(dpct::experimental::detail::json_stringstream &ss, Point2D_codepin &value,
-//CHECK:                     dpct::experimental::StreamType stream) {
-//CHECK:      ss << "{\"Type\":\"Point2D\",\"Data\":[";
-//CHECK:      ss << "{\"x\":";
-//CHECK:      dpct::experimental::detail::DataSer<int>::dump(ss, value.x, stream);
-//CHECK:      ss << "},";
-//CHECK:      ss << "{\"y\":";
-//CHECK:      dpct::experimental::detail::DataSer<int>::dump(ss, value.y, stream);
-//CHECK:      ss << "}";
-//CHECK:      ss << "]}";
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  template <> class DataSer<Point2D> {
-//CHECK:  public:
-//CHECK:    static void dump(dpct::experimental::detail::json_stringstream &ss, Point2D &value,
-//CHECK:                     dpct::experimental::StreamType stream) {
-//CHECK:      Point2D_codepin& temp = reinterpret_cast<Point2D_codepin&>(value);
-//CHECK:      dpct::experimental::detail::DataSer<Point2D_codepin>::dump(ss, temp, stream);
-//CHECK:    }
-//CHECK:  };
-
-//CHECK:  }
-//CHECK:  }
-//CHECK:  }
-//CHECK:  #endif
+//CHECK: #ifndef __DPCT_CODEPIN_GENRATED_SCHEMA__
```

Comment:
how about change the filename from generated_schema.hpp.cuda to codepin_autogen_util.hpp?   And add comment in the header:  "this file is auto-generated, introduce the main functionality of the file"